### PR TITLE
[Core] Adding quadratic geometries VTKOutput

### LINKED ISSUE #4709: vtk output problem for quadratic element
there is vtk output problem when i use structural element 3D10N instead of 3D4N. it is ok for 3D4N

Screenshot from 2019-04-21 19-51-26

### PR DESCRIPTION
**Pull request by loumalouomega**
Fixes #4709. See https://vtk.org/wp-content/uploads/2015/04/file-formats.pdf page 10

### CODE REVIEW

File: kratos/tests/test_vtk_output_process.py [hunk 1]
```diff
@@ -129,6 +135,75 @@ def SetupModelPart3D(model_part):
     bcmn.AddNodes([13, 14, 15])
     bcmn.AddConditions([3,4])
 
+def SetupModelPartQuadratic3D(model_part):
```
Comment (loumalouomega) [inline kratos/tests/test_vtk_output_process.py:138]: @ZhimingGuo here

File: kratos/tests/test_vtk_output_process.py [hunk 2]
```diff
@@ -176,8 +251,10 @@ def ExecuteBasicVTKoutputProcessCheck(file_format = "ascii", dimension = "2D"):
     model_part = current_model.CreateModelPart(model_part_name)
     if dimension == "2D":
         SetupModelPart2D(model_part)
-    else:
+    elif dimension == "3D":
```
Comment (msandre) [inline kratos/tests/test_vtk_output_process.py]: Quad should probably not be lumped in the same variable as dimension. This makes the code confusing.

Comment (loumalouomega) [reply kratos/tests/test_vtk_output_process.py]: It was the way to make the less modifications possible

Comment (loumalouomega) [reply kratos/tests/test_vtk_output_process.py]: One sec

Comment (loumalouomega) [reply kratos/tests/test_vtk_output_process.py]: I changed dimension for setup

**Comment by loumalouomega** [comment]
Done @msandre

File: kratos/tests/test_vtk_output_process.py (hunk 2 above)
Comment (msandre) [reply kratos/tests/test_vtk_output_process.py]: Thanks. That makes the variable naming less confusing.

**Comment by msandre** [comment]
@loumalouomega thanks. @adityaghantasala is it possible to test without creating as many files?

**Comment by loumalouomega** [comment]
> @loumalouomega thanks. @adityaghantasala is it possible to test without creating as many files?

It is a post-process, the creation of files is the key

**Comment by philbucher** [comment]
> @loumalouomega thanks. @adityaghantasala is it possible to test without creating as many files?

I guess for a change I am the one responsible for this :)
But as @loumalouomega points out, I don't know a better way of doing this

**Comment by philbucher** [comment]
I was kinda hoping that we won't add many more files but apparently this is not the case

Maybe after the vtk-output is more consolidated we can refactor the testing

**Comment by loumalouomega** [comment]
Any approval?

**Review by philbucher — APPROVED** [review]

**Review by msandre — APPROVED** [review]
Thanks, I thought it was at least worth asking.

**Comment by ZhimingGuo** [comment]
the problem i encountered is difference of vtk result display between element 3D4N and 3D10N. look at the attached pictures.
**the result of 3D4N(VON_MISES_STRESS 1 8703  float):**
vonmises-3D4N
**the result of 3D10N(VON_MISES_STRESS 1 9358  float):** 
vonmises-3D10N

**Comment by ZhimingGuo** [comment]
Is this extrapolation problem from gauss points to nodes? maybe now, kratos only considers extrapolation scheme for 3D4N, not other element.

**Comment by philbucher** [comment]
probably
@loumalouomega you are the expert on this, could you please check?

**Comment by loumalouomega** [comment]
The extrapolation I only tested with linear elements, but the idea underlying is the same. Should work

**Comment by loumalouomega** [comment]
Let me check, probably the number of GP is not correct

**Comment by loumalouomega** [comment]
Ok, I found that is right. The extrapolation is not working properly for quad

**Comment by loumalouomega** [comment]
There was a  missing abs, the average area may be negative for quadratic formulations

**Comment by ZhimingGuo** [comment]
Is this correct? look strange!! 
using branch core/fix-quad-extrapolation
vonmises-3D10N-2

**Comment by loumalouomega** [comment]
Can you send me the example?

**Comment by loumalouomega** [comment]
@ZhimingGuo can you send me again the example, I cannot  found the mail to continue debugging

**Pull request merged**